hal/common/critical.c

```diff
@@ -27,34 +27,34 @@
 #define EXCLUSIVE_ACCESS (!defined (__CORTEX_M0) && !defined (__CORTEX_M0PLUS))
 
 static volatile uint32_t interrupt_enable_counter = 0;
-static volatile uint32_t critical_interrupts_disabled = 0;
+static volatile bool critical_interrupts_disabled = false;
 
-static inline uint32_t get_interrupts_disabled(void)
+bool get_interrupts_disabled(void)
 {
 #if defined(__CORTEX_A9)
-    uint32_t interrupts_disabled = (__get_CPSR() & 0x80) >> 7;
+    bool interrupts_disabled = (bool)(((__get_CPSR() & 0x80) >> 7) & 0x1);
```

Comment:
c99 defines true as 0x1 and false as 0x0 so it is still safer to perform the shift , granted the &0x1 is not required though !
```
C99 §6.2.5/2: An object declared as type _Bool is large enough to store the values 0 and 1
```